python/tests/api/pyspark/experimental/test_profiler_function.py

```diff
@@ -4,25 +4,70 @@
 
 import pytest
 
+import whylogs as why
+from whylogs.api.logger.result_set import SegmentedResultSet
 from whylogs.api.pyspark.experimental import (
     collect_column_profile_views,
     collect_dataset_profile_view,
+    collect_segmented_results,
     column_profile_bytes_aggregator,
     whylogs_pandas_map_profiler,
 )
-from whylogs.core import ColumnProfileView, DatasetSchema, Resolver
+from whylogs.core import ColumnProfileView, DatasetProfile, DatasetSchema, Resolver
 from whylogs.core.metrics import StandardMetric
+from whylogs.core.segmentation_partition import SegmentFilter, segment_on_column
 from whylogs.core.stubs import pd as pd
 from whylogs.core.view.dataset_profile_view import DatasetProfileView
+from whylogs.core.view.segmented_dataset_profile_view import SegmentedDatasetProfileView
 
 TEST_LOGGER = getLogger(__name__)
 
 
+def _assert_segmented_result_sets_are_equal(results_a: SegmentedResultSet, results_b: SegmentedResultSet) -> None:
+    assert type(results_a) == type(results_b)
+
+    profiles_a = results_a.get_writables()
+    profiles_b = results_b.get_writables()
+    assert len(profiles_a) == len(profiles_b)
+    for profile_a in profiles_a:
+        assert isinstance(profile_a, SegmentedDatasetProfileView)
+        segment = profile_a.segment
+        segmented_view_a = profile_a.profile_view
+        segmented_view_b = results_b.profile(segment)
+        if isinstance(segmented_view_b, DatasetProfile):
+            segmented_view_b = segmented_view_b.view()
+        assert isinstance(segmented_view_b, DatasetProfileView)
+        if segmented_view_a is None or segmented_view_b is None:
+            assert segmented_view_a == segmented_view_b
```

Comment:
Maybe early return after this assert?

Reply:
it doesn't check timestamps or other properties, so I wanted to do the deeper checks later. But you are correct we can simplify this a bit more. I will address some of the test code issues in a follow up change to get this out.